src/metaswitch/clearwater/plugin_tests/test_chronos_plugin.py

```diff
@@ -0,0 +1,122 @@
+    staying = [constants.NORMAL_ACKNOWLEDGED_CHANGE,
+               constants.NORMAL_CONFIG_CHANGED,
+               constants.NORMAL]
+    leaving = [constants.LEAVING_ACKNOWLEDGED_CHANGE,
+               constants.LEAVING_CONFIG_CHANGED]
+
+    joining_servers = ([k for k, v in cluster_view.iteritems()
+                        if v in joining])
+    staying_servers = ([k for k, v in cluster_view.iteritems()
+                        if v in staying])
+    leaving_servers = ([k for k, v in cluster_view.iteritems()
+                        if v in leaving])
+
+    uuid_bytes = uuid.bytes
+    instance_id = ord(uuid_bytes[0]) & 0b0111111
+    deployment_id = ord(uuid_bytes[1]) & 0b00000111 
+
+    contents = dedent('''\
+        {}
+        [identity]
+        instance_id = {}
+        deployment_id = {}
+
+        [cluster]
+        localhost = {}
+        ''').format(WARNING_HEADER, instance_id, deployment_id, current_server)
+
+    for node in joining_servers:
+        contents += 'joining = {}\n'.format(node)
+    for node in staying_servers:
+        contents += 'node = {}\n'.format(node)
+    for node in leaving_servers:
+        contents += 'leaving = {}\n'.format(node)
+
+    return contents
+
+
+class TestChronosPlugin(unittest.TestCase):
+    @mock.patch('clearwater_etcd_plugins.chronos.chronos_plugin.run_command')
+    @mock.patch('clearwater_etcd_plugins.chronos.chronos_plugin.safely_write')
+    @mock.patch('metaswitch.common.alarms.alarm_manager.get_alarm')
+    def test_write_config(self, mock_get_alarm, mock_safely_write, mock_run_command):
+        """Test the chronos_plugin writes chronos cluster settings correctly"""
+
+        """Create a plugin with dummy parameters"""
+        test_uuid = uuid.uuid4()
+        plugin = ChronosPlugin(PluginParams(ip='10.0.0.1',
+                                            mgmt_ip='10.0.1.1',
+                                            local_site='local_site',
+                                            remote_site='remote_site',
+                                            remote_cassandra_seeds='',
+                                            signaling_namespace='',
+                                            uuid=test_uuid,
+                                            etcd_key='etcd_key',
+                                            etcd_cluster_key='etcd_cluster_key'))
+
+        # We expect this alarm to be called on creation of the plugin
+        mock_get_alarm.assert_called_once_with('cluster-manager', alarm_constants.CHRONOS_NOT_YET_CLUSTERED)
+
+        """Have the plugin write chronos cluster settings"""
+        # First calculate the contents of the file that should be written
+        filename = '/etc/chronos/chronos_cluster.conf'
+        cluster_view =  {"10.0.0.1": "normal", "10.0.0.2": "joining, config changed", "10.0.0.3": "joining, acknowledged change", "10.0.0.4": "leaving, config changed", "10.0.0.5": "leaving, acknowledged change"}
```

Comment:
Can you add an entry to the cluster view for each state in https://github.com/Metaswitch/clearwater-etcd/blob/dev/src/metaswitch/clearwater/cluster_manager/constants.py#L50?